cupy/sparse/compressed.py

```diff
@@ -65,6 +65,16 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
             if len(data) != len(indices):
                 raise ValueError('indices and data should have the same size')
 
+        elif base.isdense(arg1) and arg1.ndim == 2:
+            if self.format == 'csc':
```

Comment:
This branching should be delegated to the subclasses.

    m = self.dense2csx(arg1)
    self.__init__(m, shape=shape, dtype=dtype)
    return